packages/mongoose/src/decorators/index.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * © Copyright 2022 HP Development Company, L.P.
+ * SPDX-License-Identifier: MIT
+ */
+
+import { IndexOptions, SchemaOptions } from 'mongoose';
+import { decorate, DecoratorId } from '@davinci/reflector';
+import { IPropDecoratorOptions } from './types';
+
+/**
+ * Decorate a props as mongoose schema property
+ * @param options
+ */
+export function prop(options?: IPropDecoratorOptions): PropertyDecorator {
+	return decorate(
+		{
+			[DecoratorId]: 'mongoose.prop',
+			options
+		},
+		['Property', 'Method'],
+		{ allowMultiple: false, inherit: true }
+	);
+}
+
+/**
+ * Gives the ability to add compound indexes to a schema.
+ * It decorates classes
+ * @param name
+ * @param options
+ */
+// eslint-disable-next-line no-shadow
+export function index(name, options?: IndexOptions): ClassDecorator {
+	return decorate(
+		{
+			[DecoratorId]: 'mongoose.index',
+			name,
+			options
+		},
+		['Class'],
+		{ allowMultiple: false, inherit: true }
+	);
+}
+
+/**
+ * Decorate a method as:
+ * - mongoose static method is the class method is `static`
+ * - mongoose method is the class method is a `prototype` method
+ */
+export function method(): MethodDecorator {
+	return decorate(
+		{
+			[DecoratorId]: 'mongoose.method'
+		},
+		['Method'],
+		{ allowMultiple: false, inherit: true }
+	);
+}
+
+export interface IVirtualArgs {
+	ref: string; // The model to use
+	localField?: string; // Find people where `localField`
+	foreignField: string; // is equal to `foreignField`
```

Comment:
this will fix this, right? https://github.com/HPInc/davinci/issues/70

Reply:
correct, this unit test should cover it
https://github.com/HPInc/davinci/pull/136/files#diff-4e45e19270248933b5f7f75795ec340634cb4b2f2e93600f822436e474123a8fR234-R260